central/reportconfigurations/service/v2/service_impl.go

```diff
@@ -57,10 +64,89 @@ func (s *serviceImpl) PostReportConfiguration(ctx context.Context, request *apiV
 	if err != nil {
 		return nil, err
 	}
-	// TODO : Integrate with report manager when new reporting is implemented
+	// TODO ROX-16567 : Integrate with report manager when new reporting is implemented
 	// if err := s.manager.Upsert(ctx, createdReportConfig); err != nil {
 	//	 return nil, err
 	// }
 
 	return reportConfigConverter.ConvertProtoReportConfigurationToV2(createdReportConfig), nil
 }
+
+func (s *serviceImpl) UpdateReportConfiguration(ctx context.Context, request *apiV2.ReportConfiguration) (*v1.Empty, error) {
+	protoReportConfig := reportConfigConverter.ConvertV2ReportConfigurationToProto(request)
+	if err := s.validator.ValidateReportConfiguration(ctx, protoReportConfig); err != nil {
```

Comment:
`ValidateReportConfiguration` does not check ID. For updates, check for ID first.